### PR TITLE
X86: AVX2 improvements in codegen and runtime functions (by ZviRackover)

### PR DESCRIPTION
**Pull request by dsharletg**
Reopening #2680 so it gets tested by the build bots

### CODE REVIEW

**Comment by ZviRackover** [comment]
Looks like the Travis bots failed due to environment issues. Is there any way to restart the jobs or are the buildbots passes sufficient?

**Comment by steven-johnson** [comment]
Restarted travis

**Comment by ZviRackover** [comment]
Travis failed again due to environment issues. Give it another try?

**Comment by dsharletg** [comment]
I don't think we need to worry about it, the failure is unrelated and the build bots are good. This just needs a review (from someone other than me).

File: src/CodeGen_X86.cpp [hunk 1]
```diff
@@ -35,7 +35,7 @@ namespace {
 // i32(i16_a)*i32(i16_b) +/- i32(i16_c)*i32(i16_d) can be done by
 // interleaving a, c, and b, d, and then using pmaddwd. We
 // recognize it here, and implement it in the initial module.
-bool should_use_pmaddwd(Expr a, Expr b, vector<Expr> &result) {
+static bool should_use_pmaddwd(Expr a, Expr b, vector<Expr> &result) {
```
Comment (steven-johnson) [inline src/CodeGen_X86.cpp]: nit, but 'static' should be effectively a no-op inside an anon namespace. curious as to why added.

Comment (ZviRackover) [reply src/CodeGen_X86.cpp]: You're right. I probably missed the surrounding namespace with brackets.

**Pull request merged**